redis/Cargo.toml

```diff
@@ -120,9 +110,39 @@ num-bigint = []
 uuid = ["dep:uuid"]
 disable-client-setinfo = []
 
+# tls related features
+# At most one should be needed
+tls-native-tls = ["native-tls"]
+tls-rustls-native-certs = ["tls-rustls-core", "rustls-native-certs"]
+tls-rustls-webpki-roots = ["tls-rustls-core", "webpki-roots"]
+
+# async-std related features
+# At most one should be needed
+async-std-comp = ["aio", "async-std"]
+async-std-native-tls-comp = ["async-std-comp", "async-native-tls", "tls-native-tls"]
+async-std-rustls-native-certs-comp = ["async-std-rustls-core", "tls-rustls-native-certs"]
+async-std-rustls-webpki-roots-comp = ["async-std-rustls-core", "tls-rustls-webpki-roots"]
+
+# tokio related features
+# At most one should be needed
+tokio-comp = ["aio", "tokio/net"]
+tokio-native-tls-comp = ["tokio-comp", "tls-native-tls", "tokio-native-tls"]
+tokio-rustls-native-certs-comp = ["tokio-rustls-core", "tls-rustls-native-certs"]
```

Comment:
instead of having these trees of features, can we just have the basic features, with compile-time errors if additional features aren't enabled?
for example, if you enable `async-std-rustls-core`, you need to also explicitly enable either `tls-rustls-webpki-roots` or `tls-rustls-native-certs`?